docs/codeql/codeql-language-guides/analyzing-data-flow-in-swift.rst

```diff
@@ -0,0 +1,291 @@
+.. _analyzing-data-flow-in-swift:
+
+Analyzing data flow in Swift
+============================
+
+You can use CodeQL to track the flow of data through a Swift program to places where the data is used.
+
+.. include:: ../reusables/swift-beta-note.rst
+
+About this article
+------------------
+
+This article describes how data flow analysis is implemented in the CodeQL libraries for Swift and includes examples to help you write your own data flow queries.
+The following sections describe how to use the libraries for local data flow, global data flow, and taint tracking.
+For a more general introduction to modeling data flow, see ":ref:`About data flow analysis <about-data-flow-analysis>`."
+
+Local data flow
+---------------
+
+Local data flow tracks the flow of data within a single function. Local data flow is easier, faster, and more precise than global data flow. Before looking at more complex tracking, you should always consider local tracking because it is sufficient for many queries.
+
+Using local data flow
+~~~~~~~~~~~~~~~~~~~~~
+
+You can use the local data flow library by importing the ``DataFlow`` module. The library uses the class ``Node`` to represent any element through which data can flow.
+The ``Node`` class has a number of useful subclasses, such as ``ExprNode`` for expressions and ``ParameterNode`` for parameters. You can map between data flow nodes and expressions/control-flow nodes using the member predicates ``asExpr`` and ``getCfgNode``:
+
+.. code-block:: ql
+
+     class Node {
+       /**
+        * Gets this node's underlying expression, if any.
```

Comment:
Suggest simplifying, similar to Ruby:

```suggestion
        * Gets the expression that corresponds to this node, if any.
```

Reply:
Thanks.

Note to self: I should follow up with changes to the library code these snippets are quoted from as well.